### PR TITLE
Add backend event images

### LINKED ISSUE #108: Add incremental database migrations

### PR DESCRIPTION
**Pull request by jdemaeyer**
* Add migration system to the backend, including backwards-compatibility with already-initialized databases
* Add model/schema for event images

@bCyberBasti I will be out-of-town over the weekend, so if you want to start on gathering the images and there are no issues with the code feel free to merge. If we need something changed before merging I'll try to find some time to implement changes.

Closes #108.

### CODE REVIEW

**Comment by jdemaeyer** [comment]
The DeepSource stuff are non-issues as the function signatures are merely following a third party interface spec. The same is true for some issues it opened automatically because we have function signatures following Scrapy's spec. @bCyberBasti maybe you can add some ignore flags for `backend/migrations` and `backend/mj/spiders` to the DeepSource config.

@ubergesundheit Could you add the re-generated API client?

**Review by ThorbenJensen — APPROVED** [review]
changes generally look good to me, even though i am not enough of a peewee expert yet to give good feedback :-(
the good news: database migrations executes successfully on my test setup for this branch.

@jdemaeyer 
maybe check out the warning by DeepSource.
and you might want to rebase on master, to make sure the latest backend updates are compatible.

File: backend/migrations/001_initial.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"""Peewee migrations -- 001_initial.py. """
+
+import peewee as pw
+
+
+def migrate(migrator, database, fake=False, **kwargs):
+
+    @migrator.create_model
+    class EventSource(pw.Model):
```
Comment (ubergesundheit) [inline backend/migrations/001_initial.py:9]:
```suggestion
    class EventSource(pw.Model): # irgendwas
```

File: backend/migrations/002_event_images.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"""Peewee migrations -- 002_event_images.py."""
+
+import peewee as pw
+
+
+def migrate(migrator, database, fake=False, **kwargs):
```
Comment (bCyberBasti) [inline backend/migrations/002_event_images.py:6]:
```suggestion
def migrate(migrator, database, fake=False, **kwargs):  # skipcq: PYL-W0613
```

File: backend/migrations/002_event_images.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            model=migrator.orm['event'])
+        url = pw.CharField(max_length=4095)
+        description = pw.CharField(max_length=4095, null=True)
+        source = pw.CharField(max_length=1023, null=True)
+
+        class Meta:
+            table_name = "eventimage"
+
+
+def rollback(migrator, database, fake=False, **kwargs):
```
Comment (bCyberBasti) [inline backend/migrations/002_event_images.py:24]:
```suggestion
def rollback(migrator, database, fake=False, **kwargs):  # skipcq: PYL-W0613
```

File: backend/migrations/001_initial.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+"""Peewee migrations -- 001_initial.py. """
+
+import peewee as pw
+
+
+def migrate(migrator, database, fake=False, **kwargs):
```
Comment (bCyberBasti) [inline backend/migrations/001_initial.py:6]:
```suggestion
def migrate(migrator, database, fake=False, **kwargs):  # skipcq: PYL-W0613
```

File: backend/migrations/001_initial.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            field='id',
+            model=migrator.orm['organizer'],
+            null=True)
+
+        class Meta:
+            table_name = "event"
+            indexes = [(('source', 'source_event_id'), True)]
+
+
+def rollback(migrator, database, fake=False, **kwargs):
```
Comment (bCyberBasti) [inline backend/migrations/001_initial.py:69]:
```suggestion
def rollback(migrator, database, fake=False, **kwargs):  # skipcq: PYL-W0613
```

File: backend/migrations/002_event_images.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+"""Peewee migrations -- 002_event_images.py."""
+
+import peewee as pw
+
+
+def migrate(migrator, database, fake=False, **kwargs):  # skipcq: PYL-W0613
+
+    @migrator.create_model
+    class EventImage(pw.Model):
```
Comment (bCyberBasti) [inline backend/migrations/002_event_images.py:9]:
```suggestion
    class EventImage(pw.Model):  # skipcq: PYL-W0612
```

File: backend/migrations/001_initial.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
+"""Peewee migrations -- 001_initial.py. """
+
+import peewee as pw
+
+
+def migrate(migrator, database, fake=False, **kwargs):  # skipcq: PYL-W0613
+
+    @migrator.create_model
+    class EventSource(pw.Model):
```
Comment (bCyberBasti) [inline backend/migrations/001_initial.py:9]:
```suggestion
    class EventSource(pw.Model):  # skipcq: PYL-W0612
```

File: backend/migrations/001_initial.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        id = pw.AutoField()
+        name = pw.CharField(max_length=255, unique=True)
+
+        class Meta:
+            table_name = "eventsource"
+
+    @migrator.create_model
+    class Location(pw.Model):
```
Comment (bCyberBasti) [inline backend/migrations/001_initial.py:17]:
```suggestion
    class Location(pw.Model):  # skipcq: PYL-W0612
```

File: backend/migrations/001_initial.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        id = pw.AutoField()
+        description = pw.CharField(max_length=255, unique=True)
+
+        class Meta:
+            table_name = "location"
+
+    @migrator.create_model
+    class Organizer(pw.Model):
```
Comment (bCyberBasti) [inline backend/migrations/001_initial.py:25]:
```suggestion
    class Organizer(pw.Model):  # skipcq: PYL-W0612
```

File: backend/migrations/001_initial.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        id = pw.AutoField()
+        name = pw.CharField(max_length=255, unique=True)
+
+        class Meta:
+            table_name = "organizer"
+
+    @migrator.create_model
+    class Event(pw.Model):
```
Comment (bCyberBasti) [inline backend/migrations/001_initial.py:33]:
```suggestion
    class Event(pw.Model):  # skipcq: PYL-W0612
```

**Pull request merged**